### DISCUSSION TITLE
How can I make a field required in credential auth?

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#618] [Help] [answered]
**Your question**
How can I make a field required in credential auth?

**What are you trying to do**
I have my own custom username & password credential auth that is working well. The only problem is that you can submit the form without filling in the username and password fields, meaning there can be users without a username or password. Which is not what I want

**Feedback**
* did not find example in docs

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by iaincollins** [comment, marked as answer]
The [authorize](https://next-auth.js.org/configuration/providers#sign-in-with-credentials) option in the Credentials provider can be used to return null or return a rejected Promise.

You can also reject it with an `Error` object or by returning a `url` as a string.

The following are all valid ways to reject a sign in attempt from the Credentials provider:

* `Promise.resolve(null)`
* `Promise.reject(new Error('Invalid Credentials'))`
* `Promise.reject('/signin?error=invalid_credentials')`

This is documented here https://next-auth.js.org/providers/credentials
